src/framework/ui/qml/MuseScore/Ui/internal/StandardDialogPanel.qml

```diff
@@ -112,126 +107,105 @@ Rectangle {
         name: root.title + " " + root.text + " " + root.firstFocusBtn().text + " " + qsTrc("global", "Button")
     }
 
-    ColumnLayout {
-        id: content
-
-        anchors.fill: parent
-
-        property int contentHeight: childrenRect.height
-
-        spacing: 16
-
-        RowLayout {
-            id: textContent
-
-            property int contentWidth: (icon.visible ? icon.width + textContent.spacing : 0) + textColumn._preferredWidth
+    spacing: 18
 
-            Layout.preferredWidth: content.width
-            Layout.preferredHeight: childrenRect.height
+    StyledIconLabel {
+        id: icon
 
-            spacing: 28
+        Layout.alignment: Qt.AlignTop
+        Layout.preferredWidth: 48
+        Layout.preferredHeight: 48
 
-            StyledIconLabel {
-                id: icon
+        font.pixelSize: 48
+        iconCode: root.standardIcon(root.type)
 
-                Layout.alignment: Qt.AlignTop
-                Layout.preferredWidth: 48
-                Layout.preferredHeight: 48
-
-                font.pixelSize: 48
-                iconCode: root.standardIcon(root.type)
+        visible: root.withIcon && !isEmpty
+    }
 
-                visible: root.withIcon && !isEmpty
+    Column {
+        Layout.fillWidth: true
+        Layout.preferredWidth: Math.max(buttons.width, // The buttons finally determine the width
```

Comment:
Let's split this condition into several for simplification

Reply:
Thanks, good idea. That also allows me to add clearer accompanying comments :)